logstash-core/lib/logstash/patches/puma.rb

```diff
@@ -73,3 +73,19 @@ module Puma
   STDERR = LogStash::IOWrappedLogger.new(LogStash::NullLogger)
   STDOUT = LogStash::IOWrappedLogger.new(LogStash::NullLogger)
 end
+
+# JRuby (>= 9.2.18.0) added support for getsockopt(Socket::IPPROTO_TCP, Socket::TCP_INFO)
+# however it isn't working correctly on ARM64 likely due an underlying issue in JNR/JFFI.
+#
+# Puma uses the TCP_INFO to detect a closed socket when handling a request and has a dummy
+# fallback in place when Socket constants :TCP_INFO && :IPPROTO_TCP are not defined, see:
+# https://github.com/puma/puma/blob/v5.5.2/lib/puma/server.rb#L169-L192
+#
+# Remove this patch once https://github.com/elastic/logstash/issues/13444 gets resolved!
+Puma::Server.class_eval do
+  if closed_socket_supported? && ENV_JAVA['os.name'].match?(/Linux/i) && ENV_JAVA['os.arch'].eql?('aarch64')
+    def closed_socket?(socket)
+      false
+    end
+  end
+end
```

Comment:
Since the definition of `closed_socket?` is `false` if closed_socket isn't supported, maybe we can just change this to:

```suggestion
if ENV_JAVA['os.name'].match?(/Linux/i) && ENV_JAVA['os.arch'].eql?('aarch64')
  class Puma::Server
    def closed_socket?(socket)
      false
    end
  end
end
```

Reply:
fine by me with one minor adjustment to keep `class_eval` (in case the loading order changes to fail early if `Puma::Server` hasn't been loaded): 
```suggestion
if ENV_JAVA['os.name'].match?(/Linux/i) && ENV_JAVA['os.arch'].eql?('aarch64')
  Puma::Server.class_eval do
    def closed_socket?(socket)
      false
    end
  end
end
```